modules/site-health/admin-site-health.php

```diff
@@ -0,0 +1,272 @@
+								', ',
+								$this->model->get_translated_taxonomies()
+							);
+							break;
+						case 'nav_menus':
+							$current_theme = get_stylesheet();
+							foreach ( $value[ $current_theme ] as $location => $lang ) {
+								// translators: placeholder is the menu location name
+								$fields[ $location ]['label'] = sprintf( __( 'Menu: %s', 'polylang' ), $location );
+								array_walk(
+									$lang,
+									function ( &$value, $key ) {
+										$value = "$key:$value";
+									}
+								);
+								$fields[ $location ]['value'] .= implode( ' | ', $lang );
+							}
+							break;
+						case 'media':
+							array_walk(
+								$value,
+								function ( &$value, $key ) {
+									$value = "$key: $value";
+								}
+							);
+							$fields[ $key ]['label'] = '';
+							$fields[ $key ]['value'] .= implode( ',', $value );
+							break;
+						default:
+							$fields[ $key ]['label']   = $key;
+							$fields[ $key ]['value']   = implode( ', ', $value );
+							break;
+					}
+				}
+			}
+		}
+
+		$debug_info['polylang'] = array(
+			'label'    => __( 'Polylang Options', 'polylang' ),
+			'fields' => $fields,
+		);
+
+		return $debug_info;
+	}
+
+	/**
+	 * Add Polylang Languages settings to Site Health Informations tab.
+	 *
+	 * @param array $debug_info array options to display.
+	 * @return array
+	 * @since   2.8
+	 */
+	public function info_languages( $debug_info ) {
+		$languages = PLL()->model->get_languages_list();
+		$fields = array();
+		foreach ( $languages as $language ) {
+			foreach ( $language as $key => $value ) {
+				if ( empty( $value ) ) {
+					$value = '0';
+				}
+
+				if ( in_array(
+					$key,
+					$this->exclude_lang_key()
+				)
+				) {
+					continue;
+				}
+
+				$fields[ $key ]['label'] = $key;
+				$fields[ $key ]['value'] = $value;
+
+				if ( 'term_group' === $key ) {
+					$fields[ $key ]['label'] = _x( 'order', 'Order of the language in the language switcher', 'polylang' );
+				}
+			}
+			$flag = $this->get_flag( $language );
+
+			$lang_name = sanitize_title( $language->name );
+			$debug_info[ $lang_name ] = array(
+				// translators: placeholder is the language name
+				'label'  => sprintf( __( 'Language: %s', 'polylang' ), $language->name ),
+				// translators: placeholder is the flag image
+				'description' => sprintf( __( 'Flag used in the language switcher: %s', 'polylang' ), $flag ),
+				'fields' => $fields,
+			);
+		}
+		return $debug_info;
+	}
+
+	/**
+	 * Return the flag used in the language switcher
+	 *
+	 * @param object $language language object
+	 * @return mixed
+	 * @since   2.8
+	 */
+	public function get_flag( $language ) {
+		if ( empty( $language->flag ) ) {
+			return $flag = '<span class="no-flag">' . __( 'Undefined', 'polylang' ) . '</span>';
+		}
+		if ( ! empty( $language->custom_flag_url ) ) {
+			return $flag = '<img src="' . $language->custom_flag_url . '" height="11">';
+		} else {
+			return $flag = $language->flag;
+		}
+	}
```

Comment:
I don't believe you are giving a faithful information about the flag used in the switcher if you rewrite the logic inside the site health module, plus it will be two places to modify the code each time we change the flag display logic. Also, didn't we made the language code as a placeholder for unavailable flags?

Instead, [the code that displays the flag in the switcher](https://github.com/polylang/polylang/blob/0c8e04725c5810bdeed206f02c52206ba111d914/include/switcher.php#L96) should be made as a function to reuse here. This may induce additional changes because I believe that the flags will be differently set in the context of the admin dashboard than on the frontend. 

@manooweb needed to display the flags for the switcher in the admin dashboard too, so the two features may require the same change.